### ISSUE TITLE
Class variables access

### ISSUE THREAD

**Issue by riqui99** [andywer/threads.js#55]
Hello, first of all sorry for my english.

I'm using the last version of threads.js in a electron project. I'm doing something like (its only a sample.... maybe are syntax errors...):

```
function myClass(){
    var my_sub_class = new Subclass();
    const Pool = require('threads').Pool;
    const pool = new Pool();
    const _pool_method = pool.run(function(opts, done){
        my_sub_class.method(opts);
        done();
    }
    this.public_method = funciton(opts){
        _pool_method.send(opts)
        .on('done', fucntion(){ console.log('SUCCESS'); })
        .on('error', function(e){ console.log(e); })
    }
   return this;
}
var a = new myClass();
a.public_method({hello: "world"})
```

This return an error: "Error: my_sub_class.method is not a function ......"

Can I access to my local variable, that is instance of another class (my_sub_class) inside the execution of thread pool? How?

Thanks a lot!!

**Comment by andywer** [comment]
Hey @riqui99.

That's an easy one 😉
See the big comment in the [first usage example](https://github.com/andywer/threads.js#basic-usage)? The same applies to `pool.run()`.

The function you pass to `pool.run()` will actually be run in another execution context, so you are not able to access any variables (or functions, classes, ...) defined outside this function by default.

You have got to instantiate `Subclass` in the function you pass to `pool.run()`.

**Comment by riqui99** [comment]
😓Ups sorry I do not see this comment in the code, I went directly to ThreadPool docs.

It's really a pity, the instance of class that I need to pass needs "long time" to initialize, I cannot do init in every process. 😭😭

I tried to pass by parameter in the function "_pool_method" but, it does not work either.

Any idea to do something here? it's impossible no?

Thanks a lot!!!

**Comment by andywer** [comment]
It's really hard to tell without knowing what your `Subclass` actually does...

But it would probably be nice to have an easy way to pass an init function to the pool, so that it makes sure every one of its threads ran this function once before starting to do actual work. Unfortunately there is no such thing yet.

Maybe the issue originates from the design of `Subclass`, maybe it does too much? (Might not be true, I don't know what it does)

**Comment by riqui99** [comment]
My subClass only manage files to googleCloud via his official api (upload-download-modify-create biuckets...). I need to connect and create a bucket, and do this per each file is lazy because he needs a authentication.. furthermore I have a flag to change from debug to production mode, that is global in my electron project. 

But well, no problem.. I can go with synchronous, if I really need do in parallel for performance, surely I will do only a function to do an upload, not using the "Subclass".

Thanks for your comments!!!

**Comment by andywer** [comment]
You are welcome :) Hope that helped.

Will close this for now.
